hal/common/critical.c

```diff
@@ -237,6 +237,13 @@ bool core_util_atomic_cas_u32(uint32_t *ptr, uint32_t *expectedCurrentValue, uin
     return success;
 }
 
+bool core_util_atomic_cas(void **ptr, void **expectedCurrentValue, void *desiredValue) {
+    return core_util_atomic_cas_u32(
+            (unsigned *)ptr,
```

Comment:
You and @rgrover raise a good point. I'll change it to `ptrdiff_t` next chance I get ([prior art](http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith2)).